### PR TITLE
manifest: update openthread feature sets and docs

### PR DESCRIPTION
**Pull request by edmont**
Update OpenThread feature sets and their documentation.

Signed-off-by: Eduardo Montoya <eduardo.montoya@nordicsemi.no>

### CODE REVIEW

**Review by lmaciejonczyk — APPROVED** [review]

File: doc/nrf/releases/release-notes-changelog.rst [hunk 1]
```diff
@@ -58,6 +58,7 @@ Thread
 ------
 
 * Changed how Thread 1.2 is enabled, refer to :ref:`thread_ug_thread_specification_options` and :ref:`thread_ug_feature_sets` for more information.
+* Modified feature sets with support for SRP client and removal or rarely used features.
```
Comment (canisLupus1313) [inline doc/nrf/releases/release-notes-changelog.rst:61]:
```suggestion
* Modified feature sets with support for SRP client and removal of rarely used features.
```

File: doc/nrf/releases/release-notes-changelog.rst [hunk 2]
```diff
@@ -58,6 +58,7 @@ Thread
 ------
 
 * Changed how Thread 1.2 is enabled, refer to :ref:`thread_ug_thread_specification_options` and :ref:`thread_ug_feature_sets` for more information.
+* Modified feature sets with support for SRP client and removal of rarely used features.
```
Comment (wiba-nordic) [inline doc/nrf/releases/release-notes-changelog.rst:61]:
```suggestion
* Modified feature sets by adding support for SRP client and removing rarely used features.
```

**Review by wiba-nordic — CHANGES_REQUESTED** [review]

**Review by canisLupus1313 — APPROVED** [review]

**Review by LuDuda — APPROVED** [review]

File: west.yml [hunk 3]
```diff
@@ -109,7 +109,7 @@ manifest:
     - name: nrfxlib
       repo-path: sdk-nrfxlib
       path: nrfxlib
-      revision: 943bf88606f0df8b705e38182c10a8cf959dba45
+      revision: pull/624/head
```
Comment (rlubos) [inline west.yml:112]: `dfeb3041c0b896c3c0f3ffd5688263d68622e8df`

**Comment by edmont** [comment]
Rebased as the nrfxlib manifest has been updated by other PR.

**Comment by edmont** [comment]
Closing as it's included in https://github.com/nrfconnect/sdk-nrf/pull/6692

**Pull request closed without merging**